worker/src/runner/container_rt.rs

```diff
@@ -115,6 +141,13 @@ impl ContainerRuntime {
         }
     }
 
+    async fn stop_container(&self, name: &str) -> eyre::Result<()> {
+        Ok(self
+            .docker
+            .stop_container(name, None::<StopContainerOptions>)
```

Comment:
Termination is a bit more involved than a single `stop_container` call. We must give the underlying application some time to perform graceful shutdown ([example impl](https://medium.com/@dsilverdi/graceful-shutdown-in-go-a-polite-way-to-end-programs-6af16e025549)).

The general idea is that we first make a "polite" call asking the container to be stopped, [giving it a predetermined period of time](https://docs.rs/bollard/latest/bollard/container/struct.StopContainerOptions.html#structfield.t) (e.g. 20 seconds) and then observe the container's status via the `wait` call the other task is performing.

For now, I'd suggest that termination is removed from this PR since it will be a substantial undertaking, which should be separated into its own patch.

Reply:
Btw, I think we should instead use the [`KillContainer` operation](https://docs.docker.com/engine/api/v1.45/#tag/Container/operation/ContainerKill) instead, as it doesn't require us to define a deadline "after which the container is _killed_"_, unlike `StopContainer`.

Assuming that we use the kill operation instead, we can first make a kill call passing the SIGTERM status and then, after our defined graceful shutdown deadline, make a second kill call, passing SIGKILL. This is advantageous for us since after such a second call we can also emit the _container killed_ `instance::Status` message.

Reply:
You're right! the `bollard` crate offers a `option` attribute under `Docker::stop_container`, which contains the Number of seconds to wait before killing the container, allowing us to execute the graceful shutdown. 

With this in mind `kill_container` is already implemented, since no wait time is given to stop execution.

I say we change termination and add `kill_container` as termination is currently being implemented.

Reply:
Yep, the problem with `stop_container` is that we don't want Docker to kill the container after the elapsed time. We should kill the container so that we know _we_ killed it and retain control over it.

1. Call `KillContainer` passing SIGTERM.
1. If the container gracefully exited (we would know since we're `wait`ing over that container), we're done; everything ok. Report `instance::Status::Terminated`.
1. Wait for the shutdown deadline.
1. After the deadline elapses, call `KillContainer` again, but passing SIGKILL.
1. When we receive the killed status (also via the `wait` call), we report `instance::Status::Killed`.

But let's leave it to another PR.

Reply:
Wait, so this termination is incomplete then? Should I just remove it from the PR?

Reply:
Yes